src/js/techreport/tableLinked.js

```diff
@@ -21,18 +25,56 @@ class TableLinked {
       // Reset what's in the table before adding new content
       tbody.innerHTML = '';
 
+      const tableApps = content?.apps || this.pageFilters.app;
+
       // Collect the settings in an object
       const tableConfig = {
-        apps: this.pageFilters.app,
+        apps: tableApps,
         config: this.pageConfig?.[this.id]?.table,
         id: this.id,
       };
 
-      tableConfig.apps.forEach(app => {
-        const data = this.data[app];
+      const filters = new URLSearchParams(window.location.search);
+      const geo = filters.get('geo') || 'ALL';
+      const rank = filters.get('rank') || 'ALL';
+
+      // sort data
+      const sortEndpoint = component.dataset.sortEndpoint;
+      const sortMetric = component.dataset.sortMetric;
+      const sortKey = component.dataset.sortKey;
+      const client = component.dataset.client;
+
+      console.log(sortMetric);
+      if(sortMetric) {
+        this.dataArray = this.dataArray.sort((techA, techB) => {
+          // Sort techs by date to get the latest
+          const aSortedDate = techA.sort((a, b) => new Date(b.date) - new Date(a.date));
+          const bSortedDate = techB.sort((a, b) => new Date(b.date) - new Date(a.date));
+          const aLatest = aSortedDate[0];
+          const bLatest = bSortedDate[0];
+
+          // Get the correct endpoint & metric
+          const aMetric = aLatest[sortEndpoint].find(row => row.name === sortMetric);
+          const bMetric = bLatest[sortEndpoint].find(row => row.name === sortMetric);
+
+          const aValue = aMetric[client][sortKey];
+          const bValue = bMetric[client][sortKey];
+
+          return bValue - aValue > 0 ? 1 : -1;
+        });
+      }
+
+      this.dataArray.forEach(technology => {
```

Comment:
I know we talked about pagination before, but I don't remember if we talked about performance improvements to the initial API requests.

So for example if we only want to show the top 10 technologies by default and we can assume that the categories API endpoint lists the technologies in descending popularity order, we could make requests to the other API endpoints with only those first 10 technologies as request params to improve performance a bit.